editoast/src/main.rs

```diff
@@ -124,6 +124,25 @@ async fn run() -> Result<(), Box<dyn Error + Send + Sync>> {
         Commands::Search(SearchCommands::Refresh(args)) => {
             refresh_search_tables(args, pg_config).await
         }
+        Commands::Infra(subcommand) => match subcommand {
+            InfraCommands::Clone(args) => {
+                match clone_infra(args, pg_config).await {
+                    Ok(cloned_infra) => println!(
+                        "✅ Infra {} ({}) was successfully cloned",
+                        cloned_infra.name.unwrap(),
+                        cloned_infra.id.unwrap()
+                    ),
+                    Err(e) => {
+                        if e.error_type == "editoast:infra:NotFound" {
+                            println!("❌ {}", e.message);
+                            return Ok(());
+                        }
+                        return Err(e.message.into());
+                    }
+                }
```

Comment:
Could you please move this error handling inside the `clone_infra` fn ? It would be more coherent with the rest of the file